### PR TITLE
Add User-Agent when downloading image streams

### LINKED ISSUE #2601: [Bug] UriImageSourceService fails to load some images on Windows
### Steps to Reproduce

```xaml
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
       x:Class="MauiApp3.MainPage">

    <VerticalStackLayout VerticalOptions="Start">
        <Label Text="1:"/>
        <Image MaximumHeightRequest="80" 
            Source="https://upload.wikimedia.org/wikipedia/commons/thumb/b/ba/Paintedcats_Red_Star_standing.jpg/187px-Paintedcats_Red_Star_standing.jpg"/>

        <Label Text="2"/>
        <Image MaximumHeightRequest="80" 
            Source="https://upload.wikimedia.org/wikipedia/commons/f/f6/Mammuthus_exilis.jpg"/>

        <Label Text="3"/>
        <Image MaximumHeightRequest="80" 
            Source="https://upload.wikimedia.org/wikipedia/commons/thumb/e/e8/Sphinx2_July_2006.jpg/180px-Sphinx2_July_2006.jpg"/>
    </VerticalStackLayout>

</ContentPage>
```

### Expected Behavior
All images are displayed.

### Actual Behavior
Only the first one is displayed. 

If you load the image inside a browser first then WinUI is able to load the image. 

Logs:
```
Microsoft.Maui.UriImageSourceService: Warning: Unable to load image URI 'https://upload.wikimedia.org/wikipedia/commons/thumb/e/e8/Sphinx2_July_2006.jpg/180px-Sphinx2_July_2006.jpg'.

System.InvalidOperationException: Unable to load image stream.
   at Microsoft.Maui.UriImageSourceService.GetImageSourceAsync(IUriImageSource imageSource, Single scale, CancellationToken cancellationToken)

Microsoft.Maui.UriImageSourceService: Warning: Unable to load image URI 'https://upload.wikimedia.org/wikipedia/commons/f/f6/Mammuthus_exilis.jpg'.

System.InvalidOperationException: Unable to load image stream.
   at Microsoft.Maui.UriImageSourceService.GetImageSourceAsync(IUriImageSource imageSource, Single scale, CancellationToken cancellationToken)
```

### Basic Information

- Version with issue: MAUI Preview 8
- SKD: net6.0-windows10.0.19041

It works fine on Android.

### Screenshots
image

### Reproduction Link

[MauiApp3.zip](https://github.com/dotnet/maui/files/7198166/MauiApp3.zip)

### PR DESCRIPTION
**Pull request by VSadov**
### Description of Change

Fetching an image stream without setting User-Agent may run into problems with site policies. 
For example with wikimedia links it may result in:
 `403  Forbidden. Please comply with the User-Agent policy: https://meta.wikimedia.org/wiki/User-Agent_policy`

### Issues Fixed

Fixes #2601

### CODE REVIEW

**Comment by VSadov** [comment]
I am adding ".NET MAUI App" for now, but we should review whether:
- this should be configurable by the user (i.e. it is customary for bots to have "bot" in the string.
- is this a reasonable default? Perhaps we should include the name of the app?

**Comment by drasticactions** [comment]
IMO it seems kinda weird to have user agents _only_ for this one instance of an HTTP client. If I was looking at web traffic coming in and see `.NET MAUI App` only for images and nothing else? And it introduces magic behind the scenes too: If I call on a URL that requests a User Agent, that handler will work, but if I use my own HTTP Client, it will throw until I set one. If you didn't know this was happening behind the scenes, you can wonder why it worked for one and not the other.

And what if you don't want _that_ specific header for your app but one of your own? 

Having one _is_ a good idea for the reasons you said, but it feels like something a user needs more control over.


---

Also, while you didn't change add this:

https://github.com/dotnet/maui/blob/c8f0062b6094d23c13d6bc60281bdc242e670df1/src/Controls/src/Core/UriImageSource.cs#L116-L124

We're adding and disposing of a whole HTTP Client for _one_ image. Generally, AFAIK, we should be keeping one instance around instead of creating and destroying them for every one.

**Comment by mattleibow** [comment]
I wonder if we can use the configuration support for this. So during the host builder process, we can do something in there and all the clients use that?

**Comment by Redth** [comment]
I'm going to suggest we at least defer this to .NET 7.

The fact that someone wants to download an image that a HTTP server not serving content based on the lack of a user agent is outside of our control or scope and something the developer building an app with MAUI will need to work around.

**Comment by Eilon** [comment]
Yeah this change definitely doesn't look right to me because that client name is wrong 100% of the time. I think maybe we could use DI to solve this by having some default instance of HttpClient registered, or otherwise allow devs to specify other HttpClient settings via some other mechanism.

If we really did want to set a default, it should be something like the actual app's name and not some generic string, but I think we should come up with a proper solution instead of rushing something in.

**Comment by Eilon** [comment]
Some more info about registering configured HttpClient's in DI:

* https://docs.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory?view=dotnet-plat-ext-6.0
* https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-6.0

**Comment by Symbai** [comment]
> 
> Also @VSadov, what do the WinUI/WPF platforms use? Do they have a user agent, or is this some configurable value?

https://github.com/dotnet/wpf/blob/3bff49dba6a9f1d124c0d0582f9f80ee22f05202/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/BitmapDownload.cs#L193

https://github.com/dotnet/wpf/blob/3bff49dba6a9f1d124c0d0582f9f80ee22f05202/src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/WpfWebRequestHelper.cs#L54-L115

https://github.com/dotnet/wpf/blob/3bff49dba6a9f1d124c0d0582f9f80ee22f05202/src/Microsoft.DotNet.Wpf/src/Shared/MS/Win32/UnsafeNativeMethodsOther.cs#L304-L324

In short: WPF uses [ObtainUserAgentString](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/ms775114(v=vs.85)) IE API

**Comment by mattleibow** [comment]
Basically, it pretends to be IE.

I think the scenario here is to stay blank for compat and then add a config point at startup where you can override it.

**Comment by rmarinho** [comment]
This should be added as a option to configurate HttpClient on the builder

**Comment by mattleibow** [comment]
Thanks for this PR but based on the discussion here, we cannot hardcode MAUI into the user string. Instead the discussion in the issue can now be the correct way to expose this value to the options/configurations.
